### PR TITLE
Update modify_vm_iface() to support coalesce

### PR DESCRIPTION
**Pull request by chloerh**
Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
@kylazhang @yalzhang Could you please help review?

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -3505,11 +3505,13 @@ def modify_vm_iface(vm_name, oper, iface_dict, index=0, virsh_instance=virsh):
     iface_virtualport_type = iface_dict.get('virtualport_type')
     iface_filter_parameters = iface_dict.get('filter_parameters')
     iface_port = iface_dict.get('port')
+    iface_coalesce = iface_dict.get('coalesce')
     del_addr = iface_dict.get('del_addr')
     del_rom = iface_dict.get('del_rom')
     del_filter = iface_dict.get('del_filter')
     del_port = iface_dict.get('del_port')
-    del_mac = "yes" == iface_dict.get('del_mac', "no")
+    del_mac = iface_dict.get('del_mac')
```
Comment (Yingshun) [inline virttest/utils_test/libvirt.py:3513]: May I ask why we need to change this line?

Comment (chloerh) [reply virttest/utils_test/libvirt.py:3513]: @Yingshun Yes, of course. The reason that I made this change is that I have checked current test case in iface_update and found that the old implementation is not working with current test code. So I did a little change to both sides to make sure those test cases works. I guess it's a kind of fix.

**Review by yalzhang — APPROVED** [review]

File: virttest/utils_test/libvirt.py (hunk 1 above)
Comment (Yingshun) [reply virttest/utils_test/libvirt.py:3513]: Ok, got it. Thanks for your explanation!

**Review by Yingshun — APPROVED** [review]

**Review by kylazhang — COMMENTED** [review]
Others are OK for me.

File: virttest/utils_test/libvirt.py (hunk 1 above)
Comment (chloerh) [reply virttest/utils_test/libvirt.py:3513]: @kylazhang you're right! I'll update nwfilter_vm_start.

**Comment by chloerh** [comment]
@kylazhang I have submitted one: https://github.com/autotest/tp-libvirt/pull/3443

**Review by kylazhang — APPROVED** [review]
LGTM

**Pull request merged**